### DISCUSSION TITLE
What happend to assign ip address in v4.0?

### DISCUSSION THREAD

**Discussion by dejantep** [netbox-community/netbox#15986] [Q&A]
I'm just testing out demo netbox and when trying to assign ip address to an interface "assign" tab is missing.
Is it removed or am i missing something?

**Comment by candlerb** [comment]
What is the URL of the page you're looking at?

If I go to https://demo.netbox.dev/ipam/ip-addresses/372/edit/ (i.e. looking at an existing IP address) then I see:

![image](https://github.com/netbox-community/netbox/assets/44789/61a72903-036c-45c7-8022-6f0ee6bd12d4)

**Comment by dejantep** [comment]
check this link
https://demo.netbox.dev/ipam/ip-addresses/add/?interface=1754&return_url=/dcim/devices/116/interfaces/
i created test_if and want to add ip address to it. I can do this in my production version 3.7.2

**Reply by candlerb** [reply]
Sorry, I don't see the problem.

Starting at an interface, it seems you clicked the "add IP address" button. You are correctly moved to the "add IP address" page at the URL you showed, with the interface that you started from pre-selected.

Simply enter an IP address in the top box and click "Create":

![image](https://github.com/netbox-community/netbox/assets/44789/2cce826d-6b3c-4e1f-98e0-460034a1cfe1)

and you can see the IP address attached to the interface here:

https://demo.netbox.dev/dcim/interfaces/1754/

Therefore, if there's something you can do in 3.7.2 that you can't in 4.0.0, please give a complete and precise list of steps to reproduce the issue.

**Comment by dejantep** [comment]
ok. here's the screenshot from v3.7.2
![image](https://github.com/netbox-community/netbox/assets/71568639/649e10d0-0a34-490e-a762-7d2db8855786)

and here's the screenshot from v4.0
![image](https://github.com/netbox-community/netbox/assets/71568639/47a58e07-d3b9-4a7a-8176-af14bd27993f)

**Reply by candlerb** [reply]
Ah right: you want to assign an *existing* IP address to an interface, rather than create a new one, and you want to take the UI path going via Interface > Add IP Address.

I note that you can get to the old page you were talking about by changing `/add/` to `/assign/` in the URL:

https://demo.netbox.dev/ipam/ip-addresses/assign/?interface=1754&return_url=/dcim/devices/116/interfaces/

...and this page has both "Create" and "Assign" tabs, and "Create" navigates you back again.

I don't know why the "Assign" tab has gone from the Add page. Maybe there was some refactoring as part of the *"Widespread cleanup & standardization of UI components"*?

EDIT: I can't see a way to navigate in the GUI to https://demo.netbox.dev/ipam/ip-addresses/bulk-add/ either.

EDIT 2: it looks like `netbox/templates/ipam/ipaddress_edit.html` has been removed, and that's the page that included `'ipam/inc/ipaddress_edit_header.html'`. It was zapped in commit 2aaa5520

**Comment by dejantep** [comment]
ok, so it's by design.

**Reply by candlerb** [reply]
I think it's an unintended regression - otherwise they would have removed the /assign/ and /bulk-add/ pages.

**Reply by dejantep** [reply]
ok then. i'll create an issue

**Comment by dejantep** [comment]
it has been mentioned already https://github.com/netbox-community/netbox/issues/15982
